src-java/floodlight-service/floodlight-modules/src/main/java/org/openkilda/floodlight/feature/HalfSizeMetadataFeature.java

```diff
@@ -0,0 +1,32 @@
+/* Copyright 2020 Telstra Open Source
+ *
+ *   Licensed under the Apache License, Version 2.0 (the "License");
+ *   you may not use this file except in compliance with the License.
+ *   You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   Unless required by applicable law or agreed to in writing, software
+ *   distributed under the License is distributed on an "AS IS" BASIS,
+ *   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *   See the License for the specific language governing permissions and
+ *   limitations under the License.
+ */
+
+package org.openkilda.floodlight.feature;
+
+import org.openkilda.model.SwitchFeature;
+
+import net.floodlightcontroller.core.IOFSwitch;
+
+import java.util.Optional;
+
+public class HalfSizeMetadataFeature extends NoviflowSpecificFeature {
+    @Override
+    public Optional<SwitchFeature> discover(IOFSwitch sw) {
+        if (is100GbHw(sw)) {
+            return Optional.of(SwitchFeature.HALF_SIZE_METADATA);
```

Comment:
Did you checked that this limitation is actual only for WB switches? I thought that it is actual for all noviflow switches.

Reply:
Not yet (but it will be weird if all nowiswitches affected).

Reply:
I have made a test and it confirms that all nowiflow switches have truncated metadata field. I have updated feature detection.

Also, I have verified `is100GbHw` check and it works incorrectly. I will make fix for it in separate PR.